### PR TITLE
refactor: project creation

### PR DESCRIPTION
**Pull request by sunilsabatp**
refactor ->1. ordering the components of project creation and following the same stylings similar to` e.g planetCash`
                  2. define types 
                  3. changed the layout of forms

### CODE REVIEW

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**…e/projects/new-project**](https://planet-webapp-oyv4oymm3-planetapp.vercel.app/profile/projects/new-project) |
|:-:|
|<a href="https://planet-webapp-oyv4oymm3-planetapp.vercel.app/profile/projects/new-project">Screenshot of /profile/projects/new-project</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/7UnbUjQ4p2pTyUTpQXSQsif12XoOlifQhMVLsSCD1-mh6IBbGcP6W6qmuDvvAHWhKHwIbEuNuUx4Uo4Dm-qoRg.png">(view full size)</a> |







Commit 0b526c292d5937c05e221239f0edeb2402e25e6c (https://planet-webapp-oyv4oymm3-planetapp.vercel.app).

**Comment by norbertschuler** [comment]
- [ ] Trying to created a conservation project get stuck on the project media screen showing the error of the screen shot if you try to go back or skip while save & continue does nothing.

Bildschirm­foto 2023-03-25 um 12 06 32

**Comment by norbertschuler** [comment]
- [ ] the continue button on any section does not seem to work properly

**Comment by sunilsabatp** [comment]
> * [ ]  Trying to created a conservation project get stuck on the project media screen showing the error of the screen shot if you try to go back or skip while save & continue does nothing.
> 
> Bildschirm­foto 2023-03-25 um 12 06 32

I'm going to look into it

**Comment by mariahosfeld** [comment]
Please compare the size of the used buttons (and arrow elements) to other sections that have already been updated. The sizing seems to be a bit off, as the buttons are quite large.

File: src/features/user/ManageProjects/components/ProjectSpending.tsx [hunk 1]
```diff
@@ -364,51 +357,44 @@ export default function ProjectSpending({
...
-            >
-              {t('manageProjects:skip')}
-            </button>
-          </div>
+          <Button
+            onClick={() => {
+              if (uploadedFiles && uploadedFiles.length > 0) {
+                handleNext(ProjectCreationTabs.REVIEW);
+              } else {
+                setErrorMessage('Please upload  report');
```
Comment (norbertschuler) [inline src/features/user/ManageProjects/components/ProjectSpending.tsx:375]: Do we want some translated error here?

**Review by norbertschuler — APPROVED** [review]
I just tested the basic workflow (not every edge case) and it seems to work.

**Comment by mariahosfeld** [comment]
- [ ] On "Basic Details" the last toggle switch is missing/not visible.
- [ ] Fields for coordinates on map on "Basic Details" are out of alignment.
- [ ] Fields for coordinates should not be transparent, very hard to read.
- [ ] Layout indicating helper text on "Detailed Analysis" does not look good, try an opaque background to avoid it looking like crossed out
grafik
- [ ] Bug: Switching the language while creating/editing a project does not change all translations, refresh is needed
- [ ] Review page needs to be updated.

**Comment by mohitb35** [comment]
I see a client side exception after clicking the edit button on a project.

Screenshot 2023-04-26 at 3 11 49 PM (2)

Screenshot 2023-04-26 at 3 11 49 PM

**Review by mohitb35 — CHANGES_REQUESTED** [review]
I see a client side exception after clicking the edit button on a project. The same happens while trying to create a new project as well.

Screenshot 2023-04-26 at 3 14 11 PM

**Review by mariahosfeld — CHANGES_REQUESTED** [review]
- [x] Facing capitalization issue on the buttons, should be shown as it is in the ressource
Example English (should be `Back to basic details`)
grafik
Example German (should be `Zurück zu den grundlegenden Details`)
grafik
- [ ] Font size: Depending on the language, the description text in some input fields is too long
Example German
grafik
- [ ] Changing the language does not manifest immediately in all places, e.g. the lables of the `Site Owner` options

**Comment by sunilsabatp** [comment]
> * [ ]  Facing capitalization issue on the buttons, should be shown as it is in the ressource
>   Example English (should be `Back to basic details`)
>   grafik
>   Example German (should be `Zurück zu den grundlegenden Details`)
>   grafik
> * [ ]  Font size: Depending on the language, the description text in some input fields is too long
>   Example German
>   grafik
> * [ ]  Changing the language does not manifest immediately in all places, e.g. the lables of the `Site Owner` options

this issue is covered under this [https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1748](PR)

File: src/features/user/ManageProjects/StepForm.module.scss
Comment (mohitb35) [inline src/features/user/ManageProjects/StepForm.module.scss:1]: Remove styles that are no longer used. For example, `.mainContainer` and `.stepContainer`

**Comment by mohitb35** [comment]
@sunilsabatp 
1. Check that the handleNext / handleBack functionality is working correctly.
2. Add proper types to the main component props and address TS warnings.

File: pages/profile/projects/new-project.tsx [hunk 2]
```diff
@@ -61,66 +63,33 @@ export default function AddProjectType(): ReactElement {
...
-              {t('manageProjects:addNewProject')}
+      {user?.type === 'tpo' ? (
+        <DashboardView
+          title={t('manageProjects:addNewProject')}
+          subtitle={
+            <div className={'add-project-title'}>
+              {t('manageProjects:addProjetDescription')}
+              <p>
+                {t('manageProjects:addProjetContact')}{' '}
+                <span>{t('manageProjects:supportLink')}</span>
```
[the remark this answers is not in the document]
Comment (sunilsabatp) [reply pages/profile/projects/new-project.tsx:79]: @mariahosfeld what's your view point about the second point ?

[the remark this answers is not in the document]
Comment (sunilsabatp) [reply pages/profile/projects/new-project.tsx]: I'm also not sure about support link, I'll conform it from @mariahosfeld

**Comment by mohitb35** [comment]
@mariahosfeld @sunilsabatp I've made a few changes and this PR is ready to review.

**Comment by mariahosfeld** [comment]
Maybe it is just my browser, but on the basic details page, at the beginning of the lon and lat input fields I have gaps in the frame
image

**Comment by mariahosfeld** [comment]
One other thing: Please move the field for Long Term Plan up, so it is above the field for the Site Owner. This is for both, restoration and conservation problems.
Other than that everything worked fine.

File: src/features/user/ManageProjects/StepForm.module.scss [hunk 3]
```diff
@@ -7,30 +7,30 @@
...
-    left: 254px;
+    // top: -20px;
+    // left: 254px;
   }
   @include lgLaptopView {
-    left: 335px;
+    // left: 335px;
   }
-}
+} */
```
Comment (sunilsabatp) [inline src/features/user/ManageProjects/StepForm.module.scss]: could you pls clean it up, if it is not needed ?

**Review by mariahosfeld — APPROVED** [review]

**Review by mohitb35 — APPROVED** [review]

**Pull request merged**